mpqp/local_storage/save.py

```diff
@@ -0,0 +1,144 @@
+"""
+This module provides functions to insert `Job` and `Result` objects into the database, 
+ensuring proper linkage between jobs and results.
+
+Functions:
+- `insert_job`: Inserts a single `Job` into the database and returns its ID.
+- `insert_result`: Inserts a `Result` or `BatchResult` into the database. Handles compilation of jobs to avoid duplicates.
+
+"""
+
+from __future__ import annotations
+
+from mpqp.execution.connection.env_manager import get_env_variable
+from mpqp.execution.job import Job
+from mpqp.execution.result import BatchResult, Result
+from mpqp.local_storage.queries import fetch_jobs_with_job
+
+
+def insert_jobs(jobs: Job | list[Job]) -> list[int | None]:
+    """
+    Insert a `Job` into the database.
```

Comment:
Single back-ticks enable math mode in the doc, not mono-font mode, you should use double back-ticks for this.

I would also not talk in term of python objects but in terms of concepts. Since it's clear what objects are manipulated from the function's signature

Reply:
done